spec/models/miq_worker_type_spec.rb

```diff
@@ -26,4 +26,15 @@
       expect(described_class.find_by(:worker_type => old_worker_name)).to be_nil
     end
   end
+
+  describe ".worker_class_names" do
+    before { described_class.seed }
+
+    it "contains properly subclassed workers", :providers_common => true do
+      described_class.worker_class_names.each do |class_name|
+        klass = class_name.constantize
+        raise NameError, "Constant problem: expected: #{class_name}, constantized: #{klass.name}" unless klass.name == class_name
```

Comment:
This should catch the same issues we were handling at runtime but before they make it in

Reply:
Note, I don't know that this would catch all of the errors because the problems occur when the parts of the namespace are loaded out of order or not the order that's expected and Rails tries to autoload something from a partially loaded tree.

I do think this is hopefully fixed with zeitwerk if we ever get to that.

I'm fine with not doing this at runtime for 2 reasons:
1) https://github.com/ManageIQ/manageiq/pull/19400 didn't actually fix anything, it only reported the problem early instead constantizing the wrong constant name and trying to call a method on the wrong constant.
2) We haven't seen this NameError issue reported since the code was added so I'm guessing it's not a common thing and letting it fail on it's own as an exception is just as good, until we can fix this with zeitwerk.

Reply:
I recall it basically finding bugs where people didn't subclass things properly, and once those were all worked out, we never saw it again, but it was a good safety net.  That being said, safety nets are better when in specs, so I think this is good.

It's basically not possible to happen anymore because of the way the code works now, where it seeds the table with the "real" class names determined by calling `descendants`, and then the `worker_class_names` queries the table.  Even so, I like having this test in case we change something in the future.

Reply:
yeah, I edited my comment.  I meant, the move from code to spec is fine.  I am fine with it in specs as it describes our expectation.  The code removal is what I think wasn't really doing too much.  Just reporting early in the unlikely event this incorrect loading of classes occured.